net/FlatBuffers/ByteBuffer.cs

```diff
@@ -414,31 +388,26 @@ private void AssertOffsetAndLength(int offset, int length)
 #endif
         }
 
-#if UNSAFE_BYTEBUFFER
+#if ENABLE_SPAN_T
 
-        public unsafe void PutSbyte(int offset, sbyte value)
+        public void PutSbyte(int offset, sbyte value)
         {
             AssertOffsetAndLength(offset, sizeof(sbyte));
-            _buffer.Buffer[offset] = (byte)value;
+            _buffer.GetSpan(offset)[0] = (byte)value;
         }
 
-        public unsafe void PutByte(int offset, byte value)
+        public void PutByte(int offset, byte value)
         {
             AssertOffsetAndLength(offset, sizeof(byte));
-            _buffer.Buffer[offset] = value;
+            _buffer.GetSpan(offset)[0] = value;
         }
 
-        public unsafe void PutByte(int offset, byte value, int count)
+        public void PutByte(int offset, byte value, int count)
         {
             AssertOffsetAndLength(offset, sizeof(byte) * count);
+            Span<byte> span = _buffer.GetSpan(offset);
             for (var i = 0; i < count; ++i)
-                _buffer.Buffer[offset + i] = value;
-        }
-
-        // this method exists in order to conform with Java ByteBuffer standards
-        public void Put(int offset, byte value)
-        {
-            PutByte(offset, value);
+                span[i] = value;
```

Comment:
This will likely incur a bounds check on each access.  That would be eliminated if this were instead:
```C#
Span<byte> span = _buffer.GetSpan(offset, count);
for (int i = 0; i < span.Length; i++)
    span[i] = value;
```

Reply:
Good call.

Reply:
Could also be:
```C#
_buffer.GetSpan(offset, count).Fill(value);
```
Whether that's beneficial would depend on how big the span is.